CMakeLists.txt

```
@@ -21,10 +21,46 @@ cmake_minimum_required(VERSION 3.13)
 
 project(rayleigh LANGUAGES C Fortran)
 
+# load in version info and export it
+file(STRINGS "${CMAKE_SOURCE_DIR}/VERSION" RAYLEIGH_PACKAGE_VERSION LIMIT_COUNT 1)
+
+
+message(STATUS "")
+message(STATUS "====================================================")
+message(STATUS "===== Configuring Rayleigh ${RAYLEIGH_PACKAGE_VERSION}")
+message(STATUS "====================================================")
+message(STATUS "")
+
+# Set up CMAKE_BUILD_TYPE to Debug if it is not set by the user.
+if ("${CMAKE_BUILD_TYPE}" STREQUAL "")
+  set(CMAKE_BUILD_TYPE
+      "Release"
+      CACHE STRING
+      "Choose the type of build, options are: Debug, Release."
+      FORCE)
+  # Set the possible values of build type for cmake-gui
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS
+  "Debug" "Release")
```

Let's offer all the normal CMake build types. Especially `RelWithDebInfo` can be very useful.

```suggestion
      "Choose the type of build, options are: Debug, Release, RelWithDebInfo, MinSizeRel."
      FORCE)
  # Set the possible values of build type for cmake-gui
  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS
  "Debug" "Release" "RelWithDebInfo" "MinSizeRel")
```